Create function to read keys - Closes #5233

ManuGowda opened the pull request:
### What was the problem?

This PR resolves #5233 

### How was it solved?

- Implemented `readKey` function to get `fieldNumber, wireType`
### How was it tested?

- Added unit test
- Added benchmark test for performace

shuse2 commented on elements/lisk-codec/benchmark/buffer_concat.js:
```diff
@@ -0,0 +1,29 @@
+
+suite
+  .add('Buffer.concat', () => {
+    Buffer.concat([buf1, buf2]);
```
I think this is not needed? if we gonna keep this, probably nice to test with more elements and also with totalLength and without totalLength for the second parameter on the `Buffer.concat`

ManuGowda replied:
ok should i remove this now?

shuse2 replied:
Either way it's fine

ManuGowda commented:
> `keys.ts` and `readkey.spec.ts` look good to me.
> I have to admit, I didn't really understand some of the other files (like `buffer_concat.js `)

the `buffer_concat` came out of discussion we were having about performance impacts, its not relevant to this PR, so i removed it from this PR and added to #5303
